docs/en/api/S3-API.md

```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
+
+### Folders in ListObjects(V2)
+
+All sub-directories in Alluxio will be returned in ListObjects(V2) as 0-byte folders. This behavior is in accordance
+with if you used the AWS S3 console to create all parent folders for each object.
+
+### Tagging & Metadata Limits
+
+User-defined tags on buckets & objects are limited to 10 and obey the [S3 tag restrictions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html).
+
+The maximum size for user-defined metadata in PUT-requests is 2KB by default in accordance with [S3 object metadata restrictions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html).
+See the property key `alluxio.proxy.s3.header.metadata.max.size` to change this behavior.
+
+### Performance Implications
+
+The S3 API leverages the [Alluxio REST proxy]({{ '/en/api/FS-API.html#rest-api' | relativize_url }}), introducing an extra hop.
+For optimal performance, it is recommended to run the proxy server and an Alluxio worker on each compute node.
+It is also recommended to put all the proxy servers behind a load balancer.
+
+## Global request headers
+<table class="table table-striped">
+  <tr>
+    <th>Header</th>
+    <th>Content</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td><a href="https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html">Authorization</a></td>
+    <td>AWS4-HMAC-SHA256
+    Credential=<b>{user}</b>/...,
+    SignedHeaders=...,
+    Signature=...</td>
+    <td>There is currently no support for access & secret keys in the Alluxio S3 API.
+    The only supported authentication scheme is the <a href="{{ '/en/operation/Security.html#simple' | relativize_url }}">SIMPLE</a>
+    authentication type. By default, the user that is used to perform any operations is the user that was used to
+    launch the Alluxio proxy process.
+    <br/><br/>
+    Therefore this header is used exclusively to specify an Alluxio ACL username to perform an
+    operation with. In order to remain compatible with other S3 clients, the header is still
+    expected to follow the
+    <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html">AWS Signature Version 4</a>
+    format.
+    <br/><br/>
+    When supplying an access key to an S3 client, put the intended Alluxio ACL username.
+    The secret key is unused so you may use any dummy value.</td>
+  </tr>
+</table>
+
+## Supported S3 API Actions
+
+The following table describes the support status for current [S3 API Actions](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Operations.html):
 
 <table class="table table-striped">
-  <tr><th>S3 Feature</th><th>Status</th></tr>
-{% for item in site.data.table.s3-api-supported-operations %}
   <tr>
-    <td>{{ item.S3Feature }}</td>
-    <td>{{ item.Status }}</td>
+    <th>S3 API Action</th>
+    <th>REST API Endpoint</th>
```

Comment:
@Xenorith In hindsight, I think including this alone is insufficient as some endpoints are shared (with the same query parameters) but use different request bodies.

I think I should remove this column entirely (as they should look at the S3 documentation for details). wdyt?

Reply:
Removed